controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStreamMetadataStore.java

```diff
@@ -131,6 +134,19 @@ public OperationContext createStreamContext(String scopeName, String streamName,
         return new StreamOperationContext(scope, stream, requestId);
     }
 
+    @Override
+    public CompletableFuture<Void> addEntryToDeletingScope(String scope, OperationContext context, ScheduledExecutorService executor) {
+        return Futures.toVoid(createTableIfAbsent(() -> storeHelper.addNewEntry(
+                DELETING_SCOPES_TABLE, scope, UUID.randomUUID(), UUID_TO_BYTES_FUNCTION, context.getRequestId()), context)
+        );
+    }
+
+    private <T> CompletableFuture<T> createTableIfAbsent(Supplier<CompletableFuture<T>> futureSupplier,
+                                                             OperationContext context) {
```

Comment:
This method just repeats code inside `PravegaTablesScope.withCreateTableIfAbsent()` it may be better to reuse that method instead.

Reply:
same method is now being used